### PR TITLE
Chore!: bump sqlglot to v25.10.0

### PR DESCRIPTION
**Pull request by georgesittas**

### CODE REVIEW

File: tests/core/engine_adapter/test_bigquery.py [hunk 1]
```diff
@@ -653,7 +653,7 @@ def test_select_partitions_expr():
             granularity="day",
             database="{{ target.database }}",
         )
-        == "SELECT MAX(PARSE_DATE('%Y%m%d', partition_id)) FROM `{{ target.database }}.{{ adapter.resolve_schema(this) }}.INFORMATION_SCHEMA.PARTITIONS` WHERE table_name = '{{ adapter.resolve_identifier(this) }}' AND NOT partition_id IS NULL AND partition_id <> '__NULL__'"
+        == "SELECT MAX(PARSE_DATE('%Y%m%d', partition_id)) FROM `{{ target.database }}`.`{{ adapter.resolve_schema(this) }}`.INFORMATION_SCHEMA.PARTITIONS WHERE table_name = '{{ adapter.resolve_identifier(this) }}' AND NOT partition_id IS NULL AND partition_id <> '__NULL__'"
```
Comment (tobymao) [inline tests/core/engine_adapter/test_bigquery.py:656]: what happened bete

Comment (georgesittas) [reply tests/core/engine_adapter/test_bigquery.py:656]: https://github.com/tobymao/sqlglot/commit/7169e6ef52d24754059b9ee4324398d22ddff0da this commit

Comment (georgesittas) [reply tests/core/engine_adapter/test_bigquery.py:656]: I believe the quotes should be good, I ran a similar query in BQ without issues.

Comment (georgesittas) [reply tests/core/engine_adapter/test_bigquery.py:656]: The previous behavior was the weird one AFAICT.

**Pull request merged**